crates/runtime/src/cow.rs

```diff
@@ -37,7 +35,7 @@ pub struct MemoryImage {
     /// This might be an mmaped `*.cwasm` file or on Linux it could also be a
     /// `Memfd` as an anonymous file in memory. In either case this is used as
     /// the backing-source for the CoW image.
-    fd: FdSource,
+    fd: MemoryImageSource,
```

Comment:
```suggestion
    source: MemoryImageSource,
```
Maybe? Since it seems you're doing clarity refactors in this PR...